scripts/pythondistdeps.py

```diff
@@ -334,27 +373,24 @@ def normalize_name(name):
                     spec = ('==', dist.py_version)
                     if spec not in py_deps[name]:
                         py_deps[name].append(spec)
-                deps = dist.requires()
+                deps = dist.requirements
                 if args.recommends:
-                    depsextras = dist.requires(extras=dist.extras)
-                    if not args.requires:
-                        for dep in reversed(depsextras):
-                            if dep in deps:
-                                depsextras.remove(dep)
-                    deps = depsextras
+                    deps = dist.extra_requirements
                 # console_scripts/gui_scripts entry points need pkg_resources from setuptools
-                if ((dist.get_entry_map('console_scripts') or
-                    dist.get_entry_map('gui_scripts')) and
+                if (dist.entry_points and
                     (lower.endswith('.egg') or
                      lower.endswith('.egg-info'))):
-                    # stick them first so any more specific requirement overrides it
-                    deps.insert(0, Requirement.parse('setuptools'))
+                    groups = set([ep.group for ep in dist.entry_points])
+                    if set(["console_scripts", "gui_scripts"]) & groups:
+                        # stick them first so any more specific requirement
+                        # overrides it
+                        deps.insert(0, Req('setuptools'))
                 # add requires/recommends based on egg/dist metadata
                 for dep in deps:
                     if normalized_names_require_pep503:
-                        dep_normalized_name = normalize_name(dep.project_name)
+                        dep_normalized_name = normalize_name(dep.name)
                     else:
-                        dep_normalized_name = dep.key
+                        dep_normalized_name = dep.key.replace('_', '-')
```

Comment:
Why do you still do the `replace` here, but not in the other places where you use `dep.key`? I'm not saying it's necessarily wrong, but I don't understand it.

Reply:
In my testing, the previous setuptools version already has underscores replaced in the `key` property, so I would suggest doing it too.

Reply:
Be careful with manual replaces for normalized names. There are two regexes that this code supported before. The PEP 503 regext and the regex imported from setuptools ([described in here](https://github.com/pypa/setuptools/issues/1597)).

Reply:
Sorry, in here: https://github.com/pypa/setuptools/pull/1324

Reply:
The newest version looks good to me, and tests are passing.

The [documentation for pkg_resources](https://setuptools.readthedocs.io/en/latest/pkg_resources.html) says about `key`: An all-lowercase version of the project_name, useful for comparison or indexing.
From there you have to look up `project_name` and then `safe_name` function which generates it. So if we go by documentation, dots are not allowed in `key`. However, if you actually test it, they are there.

What do you think, @hroncok ?

Reply:
The code allows both naming conventions. We need to preserve that. ideally, we would have our own regexes for that to avoid changes based on 3rd party changes. We have `normalize_name()` function. What about having `normalize_name(kind="legacy"/"pep503")`?